version: '3.8'
services:
  mysql:
    image: mysql:8
    networks:
      - "mysql"
    volumes:
      - ./data:/var/lib/mysql
    ports:
      - "3306:3306"
    environment:
      - MYSQL_ROOT_PASSWORD=12345678
      - MYSQL_INIT_SQL=SET @@GLOBAL.sql_mode='STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_AUTO_VALUE_ON_ZERO,NO_ENGINE_SUBSTITUTION,NO_ZERO_DATE,NO_ZERO_IN_DATE'
    # 不知道什么原因，使用 5.7.41 的镜像加入资源限制启动不起来
    deploy:
      resources:
        limits:
          memory: 512M
    command:
      - --character-set-server=utf8mb4
      - --collation-server=utf8mb4_unicode_ci
    restart: always

networks: 
  mysql:
    external: true